### PR TITLE
Checkbox Component styling

### PR DESCRIPTION
**Pull request by zinchenkoivan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: GOOD-181

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

https://github.com/reaviz/reablocks/assets/11332713/6dae8b9e-1f1d-4c8a-865c-a5f9bcc6e042

https://github.com/reaviz/reablocks/assets/11332713/bd468a85-cd4d-48ef-a2c4-4cc1c74719f3

### CODE REVIEW

File: src/form/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -20,6 +21,11 @@ export interface CheckboxProps {
    */
   label?: string;
 
+  /**
+   * Label position of checkbox.
+   */
+  labelPosition?: 'left' | 'right';
```
Comment (amcdnl) [inline src/form/Checkbox/Checkbox.tsx:27]: Use start/end for more rtl friendly naming

```suggestion
  labelPosition?: 'start' | 'end';
```

File: src/form/Checkbox/CheckboxTheme.ts [hunk 2]
```diff
@@ -65,26 +65,60 @@ const baseTheme: Partial<CheckboxTheme> = {
...
+      'light:[&.checked]:text-vulcan',
+      'light:group-hover:text-blue-400',
+      'dark:text-waterloo',
+      'group-hover:text-blue-300',
+      'group-hover:[&.disabled]:text-waterloo',
+      'group-hover:[&.checked.disabled]:text-athens-gray',
+      '[&.checked]:text-athens-gray'
+    ].join(' ')
   },
   boxVariants: {
```
Comment (amcdnl) [inline src/form/Checkbox/CheckboxTheme.ts:117]: Didn't we just add these props?

File: src/form/Checkbox/Checkbox.story.tsx [hunk 3]
```diff
@@ -11,6 +11,30 @@ export const Simple = () => {
   return <Checkbox checked={state} label="Check me" onChange={setState} />;
 };
 
+export const WithoutLabel = () => {
+  const [state, setState] = useState(true);
+  return <Checkbox checked={state} onChange={setState} />;
+};
+
+export const LabelPosition = () => {
```
Comment (amcdnl) [inline src/form/Checkbox/Checkbox.story.tsx:19]: In the future, let's break out feature changes from styling changes for easier merging.

**Comment by zinchenkoivan** [comment]
Screenshot 2024-05-14 at 17 41 11

File: src/form/Checkbox/CheckboxTheme.ts (hunk 2 above)
Comment (zinchenkoivan) [reply src/form/Checkbox/CheckboxTheme.ts:117]: sorry, I didn't understand the question

File: src/form/Checkbox/Checkbox.story.tsx (hunk 3 above)
Comment (zinchenkoivan) [reply src/form/Checkbox/Checkbox.story.tsx:19]: it makes sense but is not always possible to break

File: src/form/Checkbox/CheckboxTheme.ts [hunk 4]
```diff
@@ -51,11 +50,12 @@ const baseTheme: Partial<CheckboxTheme> = {
...
     }
   },
-  check: '',
+  check: 'check',
   border: '',
   checkbox:
-    'flex items-center justify-center cursor-pointer focus-visible:outline-none',
-  disabled: 'opacity-60 cursor-not-allowed',
+    'checkbox flex items-center justify-center cursor-pointer focus-visible:outline-none',
+  checked: 'checked',
```
Comment (ghsteff) [inline src/form/Checkbox/CheckboxTheme.ts:57]: looks like `checked` here isn't actually applying any styling, it's just used basically as state to conditionally apply styling to the checkbox below?

It kindaaa feels like maybe this doesn't belong in the theme file cuz it's sort of part of the checkbox logic. Could `checked` be it's own set of tw classes that get applied when the `checked` prop is true in `Checkbox`? Or is there a reason you can't do it that way

e.g. might be somethin like
```
checked: {
  base: `fill-blue-400`
  disabled: `fill-gray-400`
}
```

Comment (ghsteff) [reply src/form/Checkbox/CheckboxTheme.ts:57]: what do the `check` and `checkbox` classes do?

Comment (zinchenkoivan) [reply src/form/Checkbox/CheckboxTheme.ts:57]: @ghsteff thanks for the comment, now I see what I did wrong, I do all the conditions in CheckboxTheme, although here I just need to describe the styles for all the options. I will try to move everything extra to the Checkbox component.

File: src/form/Checkbox/CheckboxTheme.ts (hunk 2 above)
Comment (zinchenkoivan) [reply src/form/Checkbox/CheckboxTheme.ts:150]: if you meant move separately "disabled, checked" then it's done

File: src/form/Checkbox/CheckboxLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  label,
+  size,
+  disabled,
+  checked,
+  onChange,
+  labelClassName,
+  theme
+}) => {
+  return (
+    <span
```
Comment (ghsteff) [inline src/form/Checkbox/CheckboxLabel.tsx:25]: can remove the unnecessary return. Im gonna add an eslint rule for this real quick 😁

**Review by ghsteff — APPROVED** [review]

File: src/form/Checkbox/CheckboxLabel.tsx (hunk 5 above)
Comment (zinchenkoivan) [reply src/form/Checkbox/CheckboxLabel.tsx]: done

File: docs/blocks/administration/Team.story.tsx [hunk 6]
```diff
@@ -75,7 +75,7 @@ export const TeamGeneral = () => {
 
   const checkboxTheme: CheckboxTheme = {
     ...defaultCheckboxTheme,
-    check: 'stroke-white',
+    checkMark: 'stroke-white',
```
Comment (amcdnl) [inline docs/blocks/administration/Team.story.tsx:78]: This is a break change - is it really nescarry ?

File: src/form/Checkbox/CheckboxTheme.ts [hunk 7]
```diff
@@ -43,57 +54,116 @@ export interface CheckboxTheme {
 }
 
 const baseTheme: Partial<CheckboxTheme> = {
-  base: 'inline-flex items-center w-full',
+  base: 'inline-flex items-center w-full group',
   label: {
-    base: 'ml-2.5 w-full',
+    base: 'text-surface-content dark:text-waterloo light:text-charade ml-2.5 w-full',
+    checked: 'checked dark:text-athens-gray light:text-vulcan',
```
Comment (amcdnl) [inline src/form/Checkbox/CheckboxTheme.ts:60]: @steppy452 - should these be referecing colors like this?

**Review by amcdnl — CHANGES_REQUESTED** [review]

File: docs/blocks/administration/Team.story.tsx (hunk 6 above)
Comment (zinchenkoivan) [reply docs/blocks/administration/Team.story.tsx:78]: not necessary,  will turn back

File: src/form/Checkbox/CheckboxTheme.ts (hunk 7 above)
Comment (zinchenkoivan) [reply src/form/Checkbox/CheckboxTheme.ts:60]: will replace normal one

**Review by amcdnl — APPROVED** [review]

**Pull request merged**